File: pset6/readability/readability.py

# Main Program


def main():
    text = input("Text: ")
    letter = count_letters(text)
    word = count_words(text)
    sentence = count_sentences(text)
    grade = cl_index(letter, word, sentence)

    # print grade
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")

# count letters


def count_letters(s):
    count = 0
    for i in range(len(s)):
        c = s[i]
        if c.isalpha() == True:
            count += 1
    return count

# count words


def count_words(s):
    count = 0
    for i in range(len(s)):
        if s[i] == ' ':
            count += 1
    return count + 1

# count sentences


def count_sentences(s):
    count = 0
    for i in range(len(s)):
        if s[i] == '.' or s[i] == '!' or s[i] == '?':
            count += 1
    return count

# Coleman-Liau index formula


def cl_index(l, w, s):
    l = float(l)
    w = float(w)
    s = float(s)
    la = 100 / w * l
    sa = 100 / w * s

    index = (0.0588 * la) - (0.296 * sa) - 15.8
    index = round(index)

    return index


# Run main program
main()

File: pset2/readability/readability.c
// Irvan Tristian
// 05 April 2020
// Problem Set 2 - Computes the Coleman-Liau index of the text

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// Prototyping function
int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
float cl_index(float l, float w, float s);

// Main Program
int main()
{
    string text = get_string("Text: ");
    int letter = count_letters(text);
    int word = count_words(text);
    int sentence = count_sentences(text);
    int grade = cl_index(letter, word, sentence);

    /*
    printf("%d letter(s)\n", letter);
    printf("%d word(s)\n", word);
    printf("%d sentence(s)\n", sentence);
    */

    // print grade
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

// count letters
int count_letters(string s)
{
    int count = 0;
    for (int i = 0, l = strlen(s); i < l; i++)
    {
        int c = (int) s[i];
        if ((c >= 65 && c <= 90) || (c >= 97 && c <= 122))
        {
            count++;
        }
    }
    return count;
}

// count words
int count_words(string s)
{
    int count = 0;
    for (int i = 0; s[i] != '\0'; i++)
    {
        if (s[i] == ' ')
        {
            count++;
        }
    }
    return count + 1;
}

// count sentences
int count_sentences(string s)
{
    int count = 0;
    for (int i = 0; s[i] != '\0'; i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            count++;
        }
    }
    return count;
}

// Coleman-Liau index formula
float cl_index(float l, float w, float s)
{
    float la = 100 / w * l;
    float sa = 100 / w * s;

    float index = (0.0588 * la) - (0.296 * sa) - 15.8;
    index = round(index);

    return index;
}